pkg/pod/status.go

```diff
@@ -229,6 +229,7 @@ func filterResultsAndResources(ctx context.Context, results []v1beta1.PipelineRe
 			cfg := config.FromContextOrDefaults(ctx)
 			aos := v1beta1.ArrayOrString{}
 			if cfg.FeatureFlags.EnableAPIFields == config.AlphaAPIFields {
+				// Note that for unsupported types such as []int will be Unmarshalled to String
```

Comment:
Can you also mention something like this "if the Type being set doesn't match what it actually should be, it will be captured somewhere else"

Reply:
Sure! You mean the validation of type (configured vs unmarshalled)? I guess that part also includes schema validation. Correct me if I'm wrong

Reply:
> Sure! You mean the validation of type (configured vs unmarshalled)? I guess that part also includes schema validation. Correct me if I'm wrong

Yes that's correct, or like schema vs object. This way we would know if the unmarshalled type matches what we expect

Reply:
Got it!